File: modules/file-transfer-core/src/main/java/com/shterneregen/securelan/filetransfer/event/FileTransferProgressEvent.java
package com.shterneregen.securelan.filetransfer.event;

import com.shterneregen.securelan.common.model.FileTransferProgress;

public record FileTransferProgressEvent(
        String transferId,
        FileTransferProgress progress,
        boolean outgoing
) implements FileTransferEvent {
}

File: modules/file-transfer-core/src/main/kotlin/com/shterneregen/securelan/filetransfer/event/FileTransferProgressEvent.kt
package com.shterneregen.securelan.filetransfer.event

import com.shterneregen.securelan.common.model.FileTransferProgress

@JvmRecord
data class FileTransferProgressEvent(
    override val transferId: String?,
    val progress: FileTransferProgress?,
    val outgoing: Boolean,
) : FileTransferEvent
